### PR TITLE
[Hub Generated] Review request for Microsoft.MarketpkaceNotifications to add version stable/2021-03-03

### PR DESCRIPTION
**Pull request by habenefr**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/habenefr/azure-rest-api-specs/dev-marketplacenotifications-Microsoft.MarketplaceNotifications-2021-03-03...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [x] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by chiragg4u** [comment]
For new services we recommend to utilize RPaaS. Please check out and use resources in the armwiki. 
Since this is a completely new RP you need to discuss the scenarios with the armapireview team first. Please send a scenario overview with how API fitin the scenario flow to the armapireview team to make progress on this.

File: specification/marketplacenotifications/resource-manager/Microsoft.MarketplaceNotifications/stable/2021-03-03/MarketplaceNotifications.json [hunk 1]
```diff
@@ -0,0 +1,256 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Microsoft Marketplace Blueprint API",
```
Comment (pilor) [inline specification/marketplacenotifications/resource-manager/Microsoft.MarketplaceNotifications/stable/2021-03-03/MarketplaceNotifications.json:4]: Are you from the Marketplace team or have you gotten explicit signoff from the marketplace team to use the Microsoft.MarketplaceNotifications namespace?

File: specification/marketplacenotifications/resource-manager/Microsoft.MarketplaceNotifications/stable/2021-03-03/MarketplaceNotifications.json [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+          }
+        },
+        "x-ms-examples": {
+          "GetReviewsNotification": {
+            "$ref": "./examples/GetReviewsNotification.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.MarketplaceNotifications/operations": {
```
Comment (pilor) [inline specification/marketplacenotifications/resource-manager/Microsoft.MarketplaceNotifications/stable/2021-03-03/MarketplaceNotifications.json:57]: must be marked x-ms-pageable  (https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi009-always-add-x-ms-pageable-to-list-calls)

File: specification/marketplacenotifications/resource-manager/Microsoft.MarketplaceNotifications/stable/2021-03-03/MarketplaceNotifications.json [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscription}/providers/Microsoft.MarketplaceNotifications/reviewsnotification": {
+      "get": {
+        "tags": [
+          "Notification"
+        ],
+        "operationId": "Notification_GetWithAuthorization",
```
Comment (pilor) [inline specification/marketplacenotifications/resource-manager/Microsoft.MarketplaceNotifications/stable/2021-03-03/MarketplaceNotifications.json:24]: The URI path for this operation is a resource type, therefore the GET should be returning a list of reviewNotifications, not a single entity.  You must also support an individual GET on each item returned in the list.  https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#get-resource

File: specification/marketplacenotifications/resource-manager/Microsoft.MarketplaceNotifications/stable/2021-03-03/MarketplaceNotifications.json [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+      "name": "api-version",
+      "in": "query",
+      "required": true,
+      "type": "string",
+      "description": "The API version to use for the request.",
+      "x-ms-parameter-location": "method"
+    }
+  },
+  "definitions": {
+    "Notification": {
```
Comment (pilor) [inline specification/marketplacenotifications/resource-manager/Microsoft.MarketplaceNotifications/stable/2021-03-03/MarketplaceNotifications.json:117]: the resource must follow the standard resource schema with a top level "id", "name", and "type" and all resource specific properties in a "properties" object.  https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#get-resource

File: specification/marketplacenotifications/resource-manager/Microsoft.MarketplaceNotifications/stable/2021-03-03/MarketplaceNotifications.json (hunk 1 above)
Comment (habenefr) [reply specification/marketplacenotifications/resource-manager/Microsoft.MarketplaceNotifications/stable/2021-03-03/MarketplaceNotifications.json:4]: yes, we are part of Marketplace team

File: specification/marketplacenotifications/resource-manager/Microsoft.MarketplaceNotifications/stable/2021-03-03/MarketplaceNotifications.json [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscription}/providers/Microsoft.MarketplaceNotifications/reviewsnotification": {
```
[the remark this answers is not in the document]
Comment (habenefr) [reply specification/marketplacenotifications/resource-manager/Microsoft.MarketplaceNotifications/stable/2021-03-03/MarketplaceNotifications.json]: changed it, and added reviewNotification get call with notificationId

**Comment by chunyu3** [comment]
Hi @habenefr 
Would you please update the PR according to the feedback of ARM otherwise the PR will be blocked to be merged?
Thanks,

File: specification/marketplacenotifications/resource-manager/Microsoft.MarketplaceNotifications/stable/2021-03-03/MarketplaceNotifications.json (hunk 4 above)
Comment (RamyasreeChakka) [reply specification/marketplacenotifications/resource-manager/Microsoft.MarketplaceNotifications/stable/2021-03-03/MarketplaceNotifications.json:188]: @habenefr - This comment is not addressed. Can you update resource schema to follow ARM standard? The resource should look like below. Id, Name, Type are at top-level properties and rest of the fields should be under properties.

{
	"id": "/subscriptions/subId/providers/Microsoft.MarketplaceNotifications/reviewsNotification/notificationId",
	"type": "Microsoft.MarketplaceNotifications/reviewsNotification",
	"name": "notificationId",
	"properties": {
		"offerId": "publisher.offer",
		"date": "02-25-2021",
		"offerDisplayName": "Offer Name",
		"principalId": "0b12389c-5e1d-4b67-880c-71d0d4a8d751"	
	}
}

File: specification/marketplacenotifications/resource-manager/Microsoft.MarketplaceNotifications/stable/2021-03-03/MarketplaceNotifications.json [hunk 6]
```diff
@@ -0,0 +1,359 @@
...
+          "type": "string",
+          "readOnly": true
+        },
+        "offerId": {
+          "description": "legacy offer id of the notification (publisher.offer)",
+          "type": "string"
+        },
+        "date": {
+          "description": "date for creating the notification",
+          "type": "string"
```
Comment (RamyasreeChakka) [inline specification/marketplacenotifications/resource-manager/Microsoft.MarketplaceNotifications/stable/2021-03-03/MarketplaceNotifications.json:213]: Specify format as "format": "date-time"

**Review by RamyasreeChakka — APPROVED** [review]
:shipit:

**Comment by chunyu3** [comment]
Hi @habenefr Please fixed the failed check first.

**Pull request merged**